hcatalog/core/src/main/java/org/apache/hive/hcatalog/cli/SemanticAnalysis/HCatSemanticAnalyzer.java

```diff
@@ -348,8 +348,8 @@ protected void authorizeDDLWork(HiveSemanticAnalyzerHookContext cntxt, Hive hive
 
     AlterTableDesc alterTable = work.getAlterTblDesc();
     if (alterTable != null) {
-      Table table = hive.getTable(SessionState.get().getCurrentDatabase(),
-          Utilities.getDbTableName(alterTable.getOldName())[1], false);
+      final String tableName = Utilities.getTableName(alterTable.getOldName()).getTable();
+      Table table = hive.getTable(SessionState.get().getCurrentDatabase(), tableName, false);
```

Comment:
`TableName` contains the database name; `SessionState.get().getCurrentDatabase()` should appear only at the time of creating the `TableName` - so as a sideeffect this "getTable()" will expect `TableName` argument

Reply:
Do you mean we should refactor getTable(String, String) to getTable(TableName)?
We can try it, but it does mean an additional change of 80 usages.